### PR TITLE
chore: update cluster-autoscaler versions

### LINKED ISSUE #741: Update cluster-autoscaler versions to fix case sensivity in resource groups
Upstream PR here:

https://github.com/kubernetes/kubernetes/pull/74882

Let's make sure we have the latest, stable cluster-autoscaler implementation for each version of k8s. E.g., 1.14 is here:

https://github.com/Azure/aks-engine/blob/master/pkg/api/k8s_versions.go#L33

Let's get the right versions configured for:

- 1.10
- 1.11
- 1.12
- 1.13
- 1.14

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

Consume new versions of cluster-autoscaler

**Issue Fixed**:

Fixes #741 

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
@feiskyer FYI

File: packer/install-dependencies.sh [hunk 1]
```diff
@@ -156,7 +156,7 @@ for TILLER_VERSION in ${TILLER_VERSIONS}; do
     echo "  - ${CONTAINER_IMAGE}" >> ${RELEASE_NOTES_FILEPATH}
 done
 
-CLUSTER_AUTOSCALER_VERSIONS="1.13.1 1.12.2 1.3.7 1.3.4 1.3.3 1.2.2 1.1.2"
+CLUSTER_AUTOSCALER_VERSIONS="1.13.1 1.13.2 1.12.2 1.12.3 1.3.7 1.3.8 1.3.4 1.3.3 1.2.2 1.2.5 1.1.2"
```
Comment (mboersma) [inline packer/install-dependencies.sh]: Could we sort these autoscaler versions in descending order? I find this confusing to read.

`1.13.2 1.13.1 1.12.3 1.12.2 1.3.8 1.3.7 1.3.4 1.3.3 1.2.5 1.2.2 1.1.2`

**Comment by mboersma** [comment]
lgtm

**Comment by CecileRobertMichon** [comment]
I had to revert this because it broke unit tests in master FYI 

```
--- FAIL: TestAssignDefaultAddonImages (11.87s)
    defaults_test.go:239: expected setDefaults to set Image k8s.gcr.io/cluster-autoscaler:v1.2.2 in addon cluster-autoscaler, but got k8s.gcr.io/cluster-autoscaler:v1.2.5
2019/03/14 01:05:37 {
```

**Comment by CecileRobertMichon** [comment]
I think this never made it back in after I reverted it, re-opening the issue and adding to backlog

**Pull request merged**